tests/tests_unit/test_api/test_functions.py

```diff
@@ -1315,3 +1317,169 @@ def validate_file_upload_call(*args, **kwargs):
+        with pytest.raises(TypeError, match=r"Arguments .* must be a subset of"):
+            _validate_function_handle(handle)
+
+    def test_validate_function_folder_with_skip_validation(self):
+        """Test that validate_function_folder passes skip_validation parameter correctly"""
+        folder = os.path.join(os.path.dirname(__file__), "function_test_resources", "function_code")
+
+        # Should not raise exception when skip_validation=True
+        validate_function_folder(folder, "handler.py", skip_folder_validation=True, skip_validation=True)
+
+        # Should also not raise exception when skip_validation=False for normal function def
+        validate_function_folder(folder, "handler.py", skip_folder_validation=True, skip_validation=False)
+
+    def test_create_function_with_skip_validation(self, mock_functions_create_response, cognite_client):
+        """Test that create method accepts skip_validation parameter"""
+        folder = os.path.join(os.path.dirname(__file__), "function_test_resources", "function_code")
+
+        # Should not raise exception
+        res = cognite_client.functions.create(
+            name="myfunction", folder=folder, function_path="handler.py", skip_validation=True
+        )
+
+        assert isinstance(res, Function)
+        assert mock_functions_create_response.calls[3].response.json()["items"][0] == res.dump(camel_case=True)
+
+    def test_create_function_with_invalid_handle_and_skip_validation_true(
+        self, mock_functions_create_response, cognite_client
+    ):
+        """Test Functions v2 / Cognite Typed Functions use case: invalid handle args with skip_validation=True"""
+
+        # Create a handle function that would normally fail validation (invalid args)
+        def handle(custom_arg1, custom_arg2, invalid_arg):
+            """This handle has invalid arguments that don't match ALLOWED_HANDLE_ARGS"""
+            return {"result": "success"}
+
+        # Should NOT raise exception when skip_validation=True
+        res = cognite_client.functions.create(name="typed_function", function_handle=handle, skip_validation=True)
+
+        assert isinstance(res, Function)
+        assert mock_functions_create_response.calls[3].response.json()["items"][0] == res.dump(camel_case=True)
+
+    def test_create_function_with_invalid_handle_and_skip_validation_false(
+        self, mock_functions_create_response, cognite_client
+    ):
+        """Test that invalid handle args fail validation when skip_validation=False"""
+
+        # Create a handle function that would fail validation (invalid args)
+        def handle(custom_arg1, custom_arg2, invalid_arg):
+            """This handle has invalid arguments that don't match ALLOWED_HANDLE_ARGS"""
+            return {"result": "success"}
+
+        # Should raise TypeError when skip_validation=False (default)
+        with pytest.raises(TypeError, match=r"Arguments .* must be a subset of"):
```

Comment:
These two can also be parametrized with `skip_validation=True` and `skip_validation=False`